### DISCUSSION TITLE
Compilation Error Tasmota 9.5

### DISCUSSION THREAD

**Discussion by smhmaqsood** [arendst/Tasmota#12932] [Q&A] [answered]
I've recently downloaded tasmota 9.5 and it is not compiling and reports an error that " 'Ping_t' was not declared in this scope."
I then moved to Tasmota 9.4 and it compiled without any error. 
- What is the possible solution to this problem?
- Can anyone guide?

**Comment by barbudor** [comment]
My crystal ball is out for maintenance.
Without more details on how you are compiling, how you got the source tree, what you may have modified/customized, build logs with the exact error message, I can just guess that you screwed something.

**Reply by ascillato2** [reply]
Arduino IDE is not supported

**Reply by smhmaqsood** [reply]
> 
> 
> My crystal ball is out for maintenance.
> Without more details on how you are compiling, how you got the source tree, what you may have modified/customized, build logs with the exact error message, I can just guess that you screwed something.

I've just downloaded the tasmota file and compile it like I used to do with previous versions. Obviously you might need your crystal ball So that you can see that it is not even compiling in PlatformIO. After maintenance of your crystal ball please let me know what is the difference between tasmota v9.4 and tasmota version v9.5 because v9.4 is compiling and v9.5 is not. 
I hope you get your crystal ball soon so that I can solve this issue.
Thanks in advance.

**Reply by barbudor** [reply]
All differences are available in GitHub
I'm compiling Tasmota almost everyday since 1 year and a half and I have no problems
Especially there have been no changes regarding compiling between 9.4.0 and 9.5.0
You should specify :
- What tool you are using : Visual Code, GitPod or TasmoCompiler
- If you are compiling a standard target without any customization or a customized target
=> If you have done a customization, please 1st try to compile tasmota.bin or tasmota32.bin without any customization to insure your setup is correct
- Provide a more complete log as copying just half a line will not help us to understand

As PlatformIO can easily screwed up itself, it is often a good point to delete `.pio` and `.cache` folders from the Tasmota tree and `.platformio` folder too (location may depend on operating system)
All missing pieces will then be redownloaded clean from scratch

**Reply by smhmaqsood** [reply]
Thanks for your guidance

**Comment by Jason2866** [comment, marked as answer]
Always use the corresponding Platformio files for the version you compile. 
Easy done by checking out from github.

**Reply by smhmaqsood** [reply]
Thanks
